pyvista/core/filters/data_set.py

```diff
@@ -1567,22 +1574,26 @@ def contour(
         # set the array to contour on
         if scalars is None:
             pyvista.set_default_active_scalars(self)
-            field, scalars = self.active_scalars_info
+            field, scalars_name = self.active_scalars_info
         else:
-            field = get_array_association(self, scalars, preference=preference)
+            field = get_array_association(self, scalars_name, preference=preference)
         # NOTE: only point data is allowed? well cells works but seems buggy?
         if field != FieldAssociation.POINT:
             raise TypeError(
                 f'Contour filter only works on Point data. Array ({scalars}) is in the Cell data.'
             )
```

Comment:
And a remark: let's clean this up for v0.36 (i.e. `preference` input kwargs vs forced point scalars, we should refactor some of these checks too probably).

Reply:
I can only see several specific instances of the general problem, e.g.

- https://github.com/pyvista/pyvista/issues/2440
- https://github.com/pyvista/pyvista/issues/2037
- https://github.com/pyvista/pyvista/issues/1856